packages/plugin-vue/src/main.ts

```diff
@@ -192,12 +192,13 @@ export async function transformMain(
   }
 
   // handle TS transpilation
-  let resolvedCode = output.join('\n')
-  if (
+  const isTs =
     (descriptor.script?.lang === 'ts' ||
       descriptor.scriptSetup?.lang === 'ts') &&
     !descriptor.script?.src // only normal script can have src
```

Comment:
tsx is also `ts` 👀

I test the case will got a error.

```html
<template>
  <h2>Ts Import</h2>
  <p class="ts-import">{{ foo }}</p>
</template>

<script lang="tsx">
import { defineComponent } from "vue"
import { foo } from './TsImportFile.js'
export default defineComponent({
  setup () {
    return {
      foo
    }
  }
})
</script>
```

Reply:
`<script lang="tsx">` isn't valid Vue, see https://sfc.vuejs.org/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8aDI+VHMgSW1wb3J0PC9oMj5cbiAgPHAgY2xhc3M9XCJ0cy1pbXBvcnRcIj57eyBmb28gfX08L3A+XG48L3RlbXBsYXRlPlxuXG48c2NyaXB0IGxhbmc9XCJ0c3hcIj5cbmltcG9ydCB7IGRlZmluZUNvbXBvbmVudCB9IGZyb20gXCJ2dWVcIlxuaW1wb3J0IHsgZm9vIH0gZnJvbSAnLi9Uc0ltcG9ydEZpbGUuanMnXG5leHBvcnQgZGVmYXVsdCBkZWZpbmVDb21wb25lbnQoe1xuICBzZXR1cCAoKSB7XG4gICAgcmV0dXJuIHtcbiAgICAgIGZvb1xuICAgIH1cbiAgfVxufSlcbjwvc2NyaXB0PiIsImltcG9ydC1tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHtcbiAgICBcInZ1ZVwiOiBcImh0dHBzOi8vc2ZjLnZ1ZWpzLm9yZy92dWUucnVudGltZS5lc20tYnJvd3Nlci5qc1wiXG4gIH1cbn0ifQ==

Only lang `ts` is supported.

Reply:
@ydcjeff linked a compiler code earlier and it looks like Vue [supports](https://github.com/vuejs/core/blob/4e7b36f5c3ee72cb9554982dd174f1989d5488f6/packages/compiler-sfc/src/compileScript.ts#L165-L169) `tsx` though, or maybe I'm mistaken 🤔

Reply:
Oh, my bad. The SFC error message is quite confusing

Reply:
Coincidentally stumbled on https://github.com/vitejs/vite/issues/2900 today which showed there used to be `App.vue?vue&type=script&lang.ts` too which would've simplified the fix. But that was changed in https://github.com/vitejs/vite/commit/93d9a2d175b1a1e3fe54197856a86887b1dadb74 as it gets inlined instead.